src/analysis/retail/priest/shadow/modules/spells/Voidform.tsx

```diff
@@ -1,39 +1,178 @@
 import SPELLS from 'common/SPELLS';
 import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
-import Events, { ApplyBuffEvent } from 'parser/core/Events';
+import Events, {
+  ApplyBuffEvent,
+  RemoveBuffEvent,
+  DamageEvent,
+  UpdateSpellUsableEvent,
+} from 'parser/core/Events';
 import Abilities from 'parser/core/modules/Abilities';
 import TALENTS from 'common/TALENTS/priest';
 import { SpellLink } from 'interface';
 import GradiatedPerformanceBar from 'interface/guide/components/GradiatedPerformanceBar';
 import { explanationAndDataSubsection } from 'interface/guide/components/ExplanationRow';
 import SpellUsable from 'parser/shared/modules/SpellUsable';
+import SPELL_CATEGORY from 'parser/core/SPELL_CATEGORY';
+
+const rate = 0.0001;
 
 class Voidform extends Analyzer {
   static dependencies = {
     abilities: Abilities,
     spellUsable: SpellUsable,
   };
-
   protected abilities!: Abilities;
   protected spellUsable!: SpellUsable;
 
-  mindblast = 0;
-  casts = 0;
+  casts = 0; //casts of voidform
+  mindblast = 0; //number of mindblasts gained by entering voidform
+
+  oncestart = 0;
+  maxCastsVB = 0; //Max casts of Voidbolt
+
+  castVB = 0; //casts of Voidbolt
+  miss = 0; //missed potential casts of Void Bolt
+
+  VB = [0]; //timestamps of voidbolt spellusable updates
 
   constructor(options: Options) {
     super(options);
 
+    this.active = this.selectedCombatant.hasTalent(TALENTS.VOID_ERUPTION_TALENT);
+
+    //I don't know how to add an event listener at the start of the fight,
+    this.addEventListener(Events.damage, this.onFightstart);
```

Comment:
you can use a listener for `combatantinfo` events i believe

Reply:
I'm still a little stuck, I tried to addEventListner for Events.combatantinfo but maybe I'm missing something I needed to import?

Reply:
try using `EventType.CombatantInfo` instead of `Events.combatantinfo`, but now that you mention it i'm not even sure that combatantinfo events get dispatched to analyzers tbh. If not, you can just leave it as you have it.

Reply:
I was using Event instead of EventType, thank you.
I gave this a try, but it didn't work. I'll keep it as is.